### PR TITLE
Revise General C++ Info

### LINKED ISSUE #1795: Improvements for MinGW Section of Running C++ Locally
It says that Mingw-w64 is somehow controlled by Microsoft, which I don't think is true. Also, the MinGW official installer section should be removed because Mingw-w64 is pretty much objectively better than it: it supports 64-bit (used by USACO) and is better at fixing bugs.

### LINKED ISSUE #2292: Encourage Use of Automatic Formatters?
Ex. https://zamhuang.medium.com/vscode-how-to-customize-c-s-coding-style-in-vscode-ad16d87e93bf

Would deal with issues such as: https://github.com/cpinitiative/usaco-guide/pull/2291/files#r739863104

### PR DESCRIPTION
**Pull request by bqi343**
- Reorganized C++ section of running code locally
- Added instructions for formatting C++ code (Closes #2292)
- Closes #1795

### CODE REVIEW

File: content/1_General/Running_Code_Locally.mdx [hunk 1]
```diff
@@ -276,288 +270,75 @@ work with the latter two. This is one of the reasons why you should **not** use
...
-		url="https://codeforces.com/blog/entry/53909"
-	 />
+	/>
 </Resources>
 
-Using this trick speeds up compilation time to less than a second!
-
-**Make sure to compile `stdc++.h` with the flags that you normally use to compile your C++ program.**
-
-#### Option 1 (Mentioned in CF blog)
```
Comment (bqi343) [inline content/1_General/Running_Code_Locally.mdx:299]: second option seems strictly better so removing this

**Review by SansPapyrus683 — APPROVED** [review]
uh sure?

File: content/1_General/Running_Code_Locally.mdx [hunk 2]
```diff
@@ -276,288 +270,76 @@ work with the latter two. This is one of the reasons why you should **not** use
...
-        `Environmental Variables` in the Windows Search Bar and click open
-    2.  Click the button `Environment Variables` at the bottom left.
-    3.  In the bottom half of the screen (under `System variables`), find the
-        key `Path` and double click.
-    4.  Click `New` on the top right and paste the path you copied. Add `\bin`
-        to the end of it. It should look something like this:
-        `C:\Program Files\mingw-w64\x86_64-8.1.0-posix-seh-rt_v6-rev0\mingw64\bin`
-    5.  You're Good to Go!
-
-#### Moderate: Microsoft Compiler Toolset
```
Comment (bqi343) [inline content/1_General/Running_Code_Locally.mdx:439]: Out of place - the other methods install some variant of `g++`. Is there a reason to prefer this method over the others? @Ta180m

File: content/1_General/Running_Code_Locally.mdx [hunk 3]
```diff
@@ -276,288 +270,76 @@ work with the latter two. This is one of the reasons why you should **not** use
...
-4.  Now, you need to add MinGW to your `Path`. To do that:
-    1.  After the Installation has completed, search for
-        `Environmental Variables` in the Windows Search Bar and click open
-    2.  Click the button `Environment Variables` at the bottom left.
-    3.  In the bottom half of the screen (under `System variables`), find the
-        key `Path` and double click.
-    4.  Click `New` on the top right and paste the path you copied. Add `\bin`
-        to the end of it. It should look something like this:
-        `C:\Program Files\mingw-w64\x86_64-8.1.0-posix-seh-rt_v6-rev0\mingw64\bin`
-    5.  You're Good to Go!
```
Comment (bqi343) [inline content/1_General/Running_Code_Locally.mdx:437]: similar instructions are included in the vscode link so doesn't seem necessary to include them here

**Comment by Aadit-Ambadkar** [comment]
Benjamin is the only person who will review his own PRs

**Review by nwatx — APPROVED** [review]

File: content/1_General/Cpp_Command.mdx [hunk 4]
```diff
@@ -184,22 +180,86 @@ on your needs. To find more information, go to
...
-		source="Benq"
-		title="My Template"
-		url="https://github.com/bqi343/USACO/blob/master/Implementations/content/contest/template.cpp"
+		source="Jetbrains"
+		title="Configuring CLion on Windows"
+		url="https://www.jetbrains.com/help/clion/quick-tutorial-on-configuring-clion-on-windows.html#MinGW"
 	>
+		Setting up MinGW with CLion
 	</Resource>
+</Resources>
```
Comment (thecodingwizard) [inline content/1_General/Cpp_Command.mdx:202]: Does mingw have support for `bits/stdc++.h`? I vaguely remember that it doesn't, but I could be wrong.

Potentially relevant links:

- https://codeforces.com/blog/entry/13981
- https://codeforces.com/blog/entry/73240

Comment (bqi343) [reply content/1_General/Cpp_Command.mdx:202]: Looks like these refer to MinGW (as opposed to Mingw-w64), though I have no idea whether there's a difference? https://github.com/cpinitiative/usaco-guide/commit/a7a14522db155a0aefcd79d48e68cdddcc0ebc12 seems to think there is.

**Review by dongliu0426 — COMMENTED** [review]
not necessary

**Pull request merged**